presto-spi/src/main/java/com/facebook/presto/spi/block/SingleRowBlock.java

```diff
@@ -114,4 +117,66 @@ public Block getLoadedBlock()
         }
         return new SingleRowBlock(rowIndex, loadedFieldBlocks);
     }
+
+    @Override
+    public byte getByteUnchecked(int position)
+    {
+        return ((UncheckedBlock) getRawFieldBlock(position)).getByteUnchecked(rowIndex);
```

Comment:
Same question here: can this be inlined to be same as raw array access?

Reply:
> How much does the lease and closing business allocate? We have the common special case where there is nothing to flatten, e.g. in the case of repartition on top of scan or hash join on top of exchange. This needs no closing, so can work with a null lease or with a singleton empty lease object. The lease often has a dynamic extent (lifetime of a code block). So this is a candidate for stack allocation. Does this actually happen? What is the allocation behavior when there are no closers? From: Timothy Meehan <notifications@github.com> Sent: Tuesday, May 28, 2019 10:34 AM To: prestodb/presto <presto@noreply.github.com> Cc: oerling <erling@xs4all.nl>; Assign <assign@noreply.github.com> Subject: Re: [prestodb/presto] Add block flattening (#12792) Assigned #12792 <#12792> to @oerling <https://github.com/oerling> . — You are receiving this because you were assigned. Reply to this email directly, view it on GitHub <#12792?email_source=notifications&email_token=AKPPPTYFOKXV6WPKMTYLUZTPXVUHVA5CNFSM4HMGZQM2YY3PNVWWK3TUL52HS4DFWZEXG43VMVCXMZLOORHG65DJMZUWGYLUNFXW5KTDN5WW2ZLOORPWSZGORVVN2OI#event-2372590905> , or mute the thread <https://github.com/notifications/unsubscribe-auth/AKPPPT2AJSUYP4P23U6KQGDPXVUHVANCNFSM4HMGZQMQ> .

When there's no closers we have some minimal object overhead, but nothing proportional in size to the underlying data.  It should never show up on the profile.